### PR TITLE
Add a settings panel

### PR DESCRIPTION
**Pull request by furbrain**
This pull request enables a settings panel for the springboard app
Pressing in the top bar of the app drops down a pane which allows you to select your wireless network and enter a password. You can also see your current IP address and there is a button to upgrade your software... but it doesn't work yet - waiting on helper script for this.

On the main screen are also now a few icons to reflect network connectivity status, keyboard, mouse and gamepad attachment.

### CODE REVIEW

**Comment by joerick** [comment]
Exciting! For the update, have a look at this https://github.com/tingbot/tingbot-os/commit/0ae8ac79e0231ebdc370e9ecd3037f062a576d25

File: springboard.tingapp/settings.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import pygame
+import tingbot
+import tingbot_gui as gui
+from icon_utils import get_network_icon_name, iconise
+
+IFACE = 'wlan0'
+
+
+def get_os_version():
+    return tingbot.__version__
```
Comment (joerick) [inline springboard.tingapp/settings.py]: This could be parsed from the output of `dpkg -s tingbot-os` or `tbupdate`

**Comment by joerick** [comment]
Hi @furbrain, what stage is this at? Ready for a review?

**Comment by furbrain** [comment]
Nearly there - I'd like to do a bit of work on it this evening - getting some error messages from the wifi library currently when connecting, so might need a little fettling. I'll link in with the tbupdate script as well

**Comment by joerick** [comment]
Great! The requirements.txt working okay? I'm going to add a 'build' command to tbtool to make a virtualenv with installed requirements so we can do that during the tingbot-os build.

**Comment by furbrain** [comment]
Requirements.txt seems to be working fine now. I would consider the settings pane to be fully functional now and ready for review.
Hint - to confuse dpkg about the current version number, edit `/var/lib/dpkg/status` ... carefully.

**Comment by joerick** [comment]
Just a note - I'm getting the following when I run:

```
Collecting tingbot-gui>=0.1 (from -r /tmp/tide/springboard.tingapp/requirements.txt (line 1))
  Could not find a version that satisfies the requirement tingbot-gui>=0.1 (from -r /tmp/tide/springboard.tingapp/requirements.txt (line 1)) (from versions: )
No matching distribution found for tingbot-gui>=0.1 (from -r /tmp/tide/springboard.tingapp/requirements.txt (line 1))
```

You'll have to upload your tingbot-gui package to PyPI or specific a Github URL in requirements.txt so that will download.

**Comment by joerick** [comment]
Great! Got it running by installing a github tarball system-wide.

**Comment by joerick** [comment]
img_5380

Here it is! Great work on this so far.

I'm gonna merge this PR and then we can work together on master of this repo. I think that'll be the easiest way to collaborate.

Here are a few issues I've encountered so far-

- Crash changing the wifi network

  > File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
  >  raise CalledProcessError(retcode, cmd, output=output)
  > CalledProcessError: Command '['/sbin/ifdown', 'wlan0']' returned non-zero exit status 1

- Crash during scrolling of wee wifi menu

  >   File "/usr/local/lib/python2.7/dist-packages/tingbot_gui/scrollarea.py", line 166, in on_touch
  >     self.start_flick(self.get_velocity(self.last_move[-1],self.last_move[0]))
  >   File "/usr/local/lib/python2.7/dist-packages/tingbot_gui/scrollarea.py", line 66, in get_velocity
  >     return [dx/dt,dy/dt]
  > ZeroDivisionError: float division by zero

- Menu system of wifi selection doesn't match designs in #1
- Keyboard can be tricky to hit buttons - sometimes the button 'down' registers (I can see the key draw depressed) but it doesn't fire when I release. Interested in your thoughts on why this might be...
- General styling - fonts, margins etc.
- Frame rate of slide transition

What would you like me help out with?  I was thinking of looking at general styling and the frame rate problem first, does that sound good to you?

**Comment by furbrain** [comment]
Would be great if you could work on general styling and frame rate. I'll look at issues 1,2 and 4 you have identified. The "missed keys" problem is a consequence of how the buttons work in the gui - they "cancel" the press if the touch moves outside of the button before being released. That is probably appropriate for larger buttons (e.g. the "Update" button, because you might change your mind at the last minute, however for the smaller buttons on the keyboard, it's probably not ideal. I'll need to fettle that in tingbot-gui.

**Comment by furbrain** [comment]
I can't seem to replicate error 1. Could you give me a bit more trace - looks like some sort of permissions error maybe in running `ifdown`?

**Comment by furbrain** [comment]
I've fixed errors 2 and 4 in tingbot-gui  fixes up-loaded to my repo

**Comment by joerick** [comment]
Merged! I've given you write permissions on this repo. Development will continue on the master branch of this repo.

**Pull request merged**